qiita_pet/handlers/analysis_handlers.py

```diff
@@ -349,3 +355,49 @@ class AnalysisSummaryAJAX(BaseHandler):
     def get(self):
         info = self.current_user.default_analysis.summary_data()
         self.write(dumps(info))
+
+
+class ShareAnalysisAJAX(BaseHandler):
+    @execute_as_transaction
+    def _get_shared_for_study(self, analysis, callback):
+        shared_links = get_shared_links(analysis)
+        users = [u.email for u in analysis.shared_with]
+        callback((users, shared_links))
+
+    @execute_as_transaction
+    def _share(self, analysis, user, callback):
+        user = User(user)
+        add_message('Analysis <a href="/analysis/results/%d">\'%s\'</a> '
+                    'has been shared with you.' %
+                    (analysis.id, analysis.name), [user])
+        callback(analysis.share(user))
+
+    @execute_as_transaction
+    def _unshare(self, analysis, user, callback):
+        user = User(user)
+        add_message('Analysis \'%s\' has been unshared from you.' %
+                    analysis.name, [user])
+        callback(analysis.unshare(user))
+
+    @authenticated
+    @coroutine
+    @execute_as_transaction
+    def get(self):
+        analysis_id = int(self.get_argument('id'))
+        analysis = Analysis(analysis_id)
+        if self.current_user != analysis.owner:
+            raise HTTPError(403, 'User %s does not have permissions to share '
+                            'analysis %s' % (
+                                self.current_user.id, analysis.id))
+
+        selected = self.get_argument('selected', None)
+        deselected = self.get_argument('deselected', None)
+
+        if selected is not None:
```

Comment:
No share or unshare actions, but the user email links and user list is still returned. This is leveraged for initial population of the shared list in the page.